[Dig] person generator: Fix Greek family names, add Polish names

GWRon opened the pull request:
Greek Names generator mistakingly had family names twice - and consisting of male first names.

This PR also adds support for Polish names.

GWRon commented:
Sample output based on
```
SuperStrict
Framework Brl.StandardIO
Import "../source/Dig/base.util.persongenerator.bmx"

GetPersonGenerator().Initialize()


For Local country:String = eachin GetPersonGenerator().GetCountryCodes()
	For local gender:Int = EachIn [1, 2]
		print "-------"
		if gender = 1
			print "country: " + country + " - male"
		else
			print "country: " + country + " - female"
		endif
		print "-------"
		For local i:int = 0 to 10
			print GetPersonGenerator().GetFirstName(country, gender) + " " + GetPersonGenerator().GetLastName(country, gender)
		Next
	Next
	print "-------"
Next
```

```
-------
country: aut - male
-------
Andreas Seidel
Emile Seidel
Florentin Christoph
Lorenzo Dippel
Umberto Henschel
Kevin Hande
Camilo Klotz
Domingo Stolze
Cian Butte
Niklas Gehringer
Andrijan Hentschel
-------
country: aut - female
-------
Daniella Vollbrecht
Albertine Müller
Giuseppa Kabus
Elsa Schulz
Marla Holt
Alexia Schüler
Brigitta Karz
Katherine Bohnbach
Terry Fliegner
Holli Kranz
Käthe Köhler
-------
-------
country: cn - male
-------
Nan Shu
Yi Gu
Wei Zhu
Asahi Ho
Bo Jia
Bo Yan
Xue-Ming Yao
Jianping Chu
Zhixin Zou
Tao Ho
Bo Tao Lee
-------
country: cn - female
-------
Xinyu Yan
Yuhua Gu
Yumei Ling
Dongmei King
Guirong Deng
Xia Cao
Guizhen Tu
Xinyu Geng
Xiurong Zhu
Lihua Ho
Yuzhen Zheng
-------
-------
country: d - male
-------
Wilhelm Ebert
Heinz-Günther Seifert
Klaus Peter Heintze
Jonas Krause
Bert Bruder
Gunnar Bachmann
Gordon Schenk
Arndt Adolph
Theodor Neureuther
Erwin Schmiedt
Antonio Etzler
-------
country: d - female
-------
Mirja Hendriks
Sigrid Bien
Emmy Dietz
Claire Reising
Walburga Sager
Malgorzata Scholtz
Jeannine Dobes
Mariana Nette
Teresa Trupp
Cäcilie Jähn
Nadeshda Sölzer
-------
-------
country: de - male
-------
Cornelius Henschel
Marc Liebelt
Phillip Beyer
Hans-Theo Boucsein
Vincent Speer
Ingolf Heuser
Hans-Albert Faust
Berndt Pieper
Hansjörg Renner
Raymond Hahn
Harald Kade
-------
country: de - female
-------
Monica Bruder
Concetta Reinhardt
Katalin Hauffer
Lidija Mans
Resi Rosemann
Britta Anders
Natascha Meister
Erna Ruppersberger
Kirstin Steckel
Mariola Gieß
Slavica Wiek
-------
-------
country: dk - male
-------
Ingemann Larsen
Evald Bjørn
Alfred Johansen
Ahmed Lynge
Rudolf Bentzen
Eigil Skøtt
Tobias Hermansen
Kresten Frandsen
Carl Christiansen
Sven Hansen
Christian Krarup
-------
country: dk - female
-------
Alberte Ottesen
Thi Kristiansen
Margit Bjerrum
Jytte Therkelsen
Jeanne Kofoed
Therese Andersen
Ella Broberg
Martha Kirkegaard
Line Bro
Christa Larsen
Sonja Brogaard
-------
-------
country: es - male
-------
Raul Amador Salvador
Gael Trujillo Nájera
Miguel Lugo Valle
Asier Collado Ureña
Jose Muñoz
Francisco Javier Sauceda
Jose Manuel Botello Ornelas
Carlos Sandoval Calvo
Omar Grijalva Ojeda
Rayan Campos Saiz
Biel Dueñas Montemayor
-------
country: es - female
-------
Berta Martín Paredes
Ines Baeza Benito
Noelia Ibáñez Millán
Isabel Villegas Esparza
Martina Galán Cordero
Carmen Villalba
Pau Gracia Razo
Gabriela Hernando Riera
Nayara Bañuelos
Sara Linares Lemus
Alicia Posada Expósito
-------
-------
country: fr - male
-------
Robin Fraise
Timéo Fontaine
Sofiane Aubert
Guillaume Travers
Jean Toussaint
Quentin Gay
William Ozanne
Matheo Monet
Esteban Toussaint
Mael Jullien
Olivier Vachon
-------
country: fr - female
-------
Jeanne Joubert
Lucile Danzas
Olivia Riviere
Emma Sartre
Rose Hector
Olivia Ozanne
Clémence Richard
Louna Husson
Lylou Monet
Mélissa Visage
Maelys Garnier
-------
-------
country: gr - male
-------
Kefalogiannis Paraskevas
Polychronis Stefanidis
Spanos Kyriazis
Oikonomou Sikinos
Skalidis Vasilakis
Anagnostidis Karas
Stelios Antonopoulos
Nikolopoulos Karas
Isidoros Nikolos
Kaloudis Makris
Kotsis Loukakis
-------
country: gr - female
-------
Katerina Kritikos
Timothea Georgiadis
Eleftheria Kondylis
Polina Makris
Asimina Papadopoulos
Emmanouela Tzimas
Argyro Marinos
Eleni Lykos
Kleopatra Giannakopoulos
Tatiana Zervas
Alexandra Daskalopoulos
-------
-------
country: pl - male
-------
Oskar Pietrzak
Mateusz Witkowski
Arkadiusz Chrzanowski
Wiktor Walczak
Michał Malinowski
Franciszek Wojciechowski
Marcel Kowalczyk
Jacek Mikulski
Aleksander Marciniak
Franciszek Nowicki
Piotr Antczak
-------
country: pl - female
-------
Aneta Chrzanowski
Małgorzata Adamski
Katarzyna Wiśniewski
Grażyna Brzeziński
Gabriela Dudek
Weronika Wróbel
Dorota Głowacki
Teresa Szczepański
Zofia Adamczyk
Dominika Marciniak
Wiktoria Dąbrowski
-------
-------
country: ru - male
-------
Igor Ustinow
Timofej Kotow
Wsewolod Orechow
Stefan Emeljanow
Wladislaw Isaew
Fjodor Lasarew
Adam Noskow
Apollon Merkuschew
Lawrentij Filatow
Adam Kulagin
Nasar Mischin
-------
country: ru - female
-------
Tamara Selesnjow
Irina Terentew
Alla Panfilow
Sinaida Aleksandrow
Olga Wasilew
Uljana Komissarow
Jana Rusakow
Nina Kabanow
Ljubow Woronow
Wera Fomin
Darja Schaschkow
-------
-------
country: tr - male
-------
Ahmet Polat
Mert Beşok
Mert Babacan
Çınar Mayhoş
Meriç Atakol
Kutay Limoncuoğlu
Berkay Sadıklar
Berke Çetin
Yağız Durmaz
Utku Özdenak
Göktürk Tuğlu
-------
country: tr - female
-------
Ada Balcı
Rüya Türkdoğan
Esma Gümüşpala
Sinem Ekşioğlu
Rüya Karabulut
Sinem Akal
Ebru Ayaydın
Ümran Atakol
Rüya Numanoğlu
Esma Köylüoğlu
Ada Çetiner
-------
-------
country: ug - male
-------
Ernest Bisheko
Taylor Kawuki
Mathew Asiimwe
Tyson Munyagwa
Felix Mukalazi
Maurice Badru
Gilbert Kirabira
Johnson Kijjoba
Edwin Aisu
Ambrose Lunyoro
Mike Musinguzi
-------
country: ug - female
-------
Mariana Mwesige
Annette Ninsiima
Sharon Bisheko
Pauline Twasiima
Trisha Atwine
Vivian Ninsiima
Eva Kakooza
Teresa Tendo
Jacklyn Kanshabe
Miracle Asasira
Margaret Bugala
-------
-------
country: uk - male
-------
Adrian Marshall
Paul Khan
Henry Richardson
Chris Chapman
Barry Williams
Jonathan Owen
Owen Khan
Alex Martin
Dean Murray
Darren Griffiths
Mike Walsh
-------
country: uk - female
-------
Emily Owen
Rebecca Williams
Grace Davies
Helena Thompson
Tara Marshall
Eden Hunter
Carrie Kennedy
Isobel Marshall
Charlotte Reid
Faye Williams
Shannon Wood
-------
-------
country: us - male
-------
Jaylin Braun
Okey Nicolas
Alfred Ullrich
Vincent Miller
Carson Frami
Florencio Mueller
Ike Stark
Bennie Howe
Demond Barton
Delmer Hagenes
Amos Fritsch
-------
country: us - female
-------
Althea Jakubowski
Vanessa Hessel
Kyla Mertz
Antoinette Feil
Mae Kris
Joy Schowalter
Carmella Von
Reba Mills
Iva Willms
Shanon Gutkowski
Aida Hagenes
-------
```

GWRon commented:
Man sieht, dass wir für "Deutschland" schon "gemischte" Namen drin haben. Also Namen der Menschen die einfach in Deutschland wohnen. Sprich man hat auch mal einen polnischen Nachnamen (vielleicht ohne "Sonderzeichen") drin usw.

Das hat man bei der USA sicher auch schon so drin - andere Laender hingegen fehlt dies evtl.

pauldim commented:
This is correct for female names:

Aneta Chrzanowska
Małgorzata Adamska
Katarzyna Wiśniewska
Grażyna Brzezińska
Gabriela Dudek
Weronika Wróbel
Dorota Głowacka
Teresa Szczepańska
Zofia Adamczyk
Dominika Marciniak
Wiktoria Dąbrowska

For male names it is ok.

GWRon commented:
So for female names the ending of the family names changes?

Is there a rule for it (can I simply suffix -ski or -ska depending on the genre)?

pauldim commented:
If the surname ends in -ski for guys then the woman will be -ska.

GWRon commented:
Are there family names which do not follow these rules?  Something being the same for all?
What about imported names? So what about someone married to a Herr Mueller?

You might see the changes in the PR and thus also see what family names I added.

Of course it is rather easy to add -ski and -ska to all the there provided family "basenames" then (we store the base then and add the suffix on name creation depending on gender).

pauldim commented:
If it can't be done easily, he suggests universal names that fit men and women:
Dudek
Wróbel
Nowak
Marciniak
Adamczyk
Kowal
Buzek
Lisek
Lasota
Bugaj
Dąbek
Machaj
Kula, Burkiewicz, Bednar, Czyż, Dolata, Kania, Ponita, Rybus, Turek, Stoch, Stokłosa, Stenka, Widera, Wrona, Zych.

GWRon commented:
As said it is is easy if you have

group1: family names in which I simply append "ski" for men and "ska" for women
group2: family names which do not need an appendix / work for both.


For now the generator knows these names:
```
		self.lastNames = [ ..
			"Adamczyk", "Adamski", "Andrzejewski", "Antczak", ..
			"Bąk", "Baran", "Baranowski", "Bartkowiak", "Bartosz", "Bednarek", "Białas", "Bielecki", "Błaszczyk", "Brzeziński", ..
			"Chrzanowski", "Czarnecki", "Czerwiński", ..
			"Dąbrowski", "Duda", "Dudek", "Dziedzic", ..
			"Gajewski", "Głowacki", "Górecki", "Grabowski", "Grzelak", ..
			"Jabłoński", "Jagielski", "Janiak", "Jankowski", "Jasiński", "Jaworski", "Jóźwiak", ..
			"Kaczmarek", "Kalinowski", "Kamiński", "Kania", "Kaczmarczyk", "Kowalczyk", "Kowalik", "Krawczyk", "Kubiak", "Kucharski", "Kwiatkowski", ..
			"Lis", ..
			"Maciejewski", "Malinowski", "Marciniak", "Mazurek", "Michalak", "Michalski", "Mikołajczyk", "Mikulski", "Mazurek", ..
			"Nowak", "Nowicki", "Nowakowski", ..
			"Pawlak", "Pietrzak", "Piotrowski",  "Puchalski", ..
			"Rutkowski", ..
			"Sadowski", "Sawicki", "Sikora", "Sobczyk", "Sokołowski", "Stępień", "Szczepański", "Szewczyk", "Szymański", ..
			"Walczak", "Wasilewski", "Wesołowski", "Wiśniewski", "Witkowski", "Wójcik", "Wojciechowski", "Woźniak", "Wrona", "Wróbel", ..
			"Zając", "Zakrzewski", "Zawadzki", "Zieliński", "Zając", "Żak", "Żuk" ..
		]
```
You could - if you want - split it into "ski/ska" and "for all". You might even extend them (eg so one of them is your family name :))

GWRon commented:
Updated generator and added your family names too

```
country: pl - male
-------
Oskar Grzelak
Bartosz Kwiatkowski
Patryk Janiak
Łukasz Kowal
Krzysztof Marciniak
Jakub Antczak
Konrad Kubiak
Patryk Lisek
Maciej Baran
Aleksander Buzek
Fabian Lasota
-------
country: pl - female
-------
Małgorzata Rutkowska
Beata Bielecka
Dominika Sobczyk
Weronika Dąbrowska
Dorota Wróbel
Teresa Zając
Zofia Zając
Kornelia Chrzanowska
Agnieszka Stoch
Celina Wójcik
Blanka Mikołajczyk
-------
```

nittka commented:
Do you intend to replace the Greek names? Currently, the new version would have far fewer names and none of them seem to have accents.
For non-exotic first characters, I would require at least 5 names for both male and female.

GWRon commented:
I am not sure if we should use the accents at all (except for the ascii-known ones).

For Chinese we also already use the "transcripts".


Regarding Greek. the old version did not contain family names (but male first names in the wrong array). Also the names did sound rather uncommon / ancient.

I am not sure how to approach that properly. In some quick shots I did not find the pages again from where the old data was sourced from.

GWRon commented:
Next to the "accents" I think Greek also has "female endings" for surnames ...
https://en.wikipedia.org/wiki/Category:Greek-language_surnames
https://en.wikipedia.org/wiki/Category:Greek-language_female_forms_of_surnames

eg. (male) Alexopoulos - Alexopoulou (female)

https://surnames.behindthename.com/names/usage/greek
![image](https://github.com/TVTower/TVTower/assets/2625226/4815ab7a-bd54-4035-8b7e-2c65448b1820)


(I will split the PR at the end into the various languages ... to keep the commits rather clean then)

GWRon commented:
Another note.

These "genitive vs female/male" variants in different languages: they often are only used "inside" the country. So the "female version" of a Greek womens name is used there, but when she applies for a job in the UK she uses the nominative/gentive one (as far as I understood).

Accents: You know ... when checking a movie's cast you will see the "transliterated" versions of the names. I assume the people who read the "names" are not all "Greek" so we need to find a consent about "naming schemes". I am not against an "accent" here and there - so French ones (é è , ë...) and also Skandinavian ones (ø) as people still manage to "read" them (albeit they might pronounce it wrong). Same for the Ł in the Polish name Łukasz. So accents in Greek names "could" be kept too.
Yet you will most probably not see it in the cast rolls of English movies - while I could imagine "arthouse" movies done in the EU pay more attention to "heritage" etc.

Where do we use the names at all?
We could use them in "news" (I think most news I read here in "German" use transliterated variants of the names, so remove a lot accents etc). We could use them in custom productions - where also a "less distracting look" (for most people a lot of accents in a word make it look... "foreign" - what it is, but also keeps their eyes "stop on it").

Just found this - maybe it could be of use somehow (sources it uses etc). But the sources there reveal how "few" the author there only knows about names in other countries (eg the first names of "Germany" consist of the most favoured baby names in the last years ... so you will not find any "Thomas" or "Andreas" but only "Noah", "Ben" etc -- so only names of people still being in school age) 
https://github.com/ozdemirburak/full-name-generator/

Dropping another link:
https://www.quora.com/How-do-masculine-and-feminine-Greek-surnames-work


@scr0llbaer9000 
I think this discussion is something you could participate in too. Your opinion for multilinguag stuff is also important.

scr0llbaer9000 commented:
I can only recomment this classic:

https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

`;)`

Nice that some effort is put into this, but I guess one can never reach "perfection" about these things.

Btw there could be some `gender 3` option if gender-neutral or -flexible names are required, I made these by hand recently:

		<firstname1> <!-- all these are more or less gender-neutral/-flexible first names, source: internet. TODO: maybe the Bavarians and the nuns need special treatment -->
			<de>Addison|Alex|Anouk|Anuk|Avery|Benja|Beverly|Billie|Bozan|Cameron|Charlie|Cherry|Celeste|Conny|Daniele|Darian|Devin|Eden|Elliot|Elya|Esra|Felice|Finley|Flo|Gale|Gerritt|Glenn|Günes|Hadir|Harley|Henny|Ihsan|Immi|Indra|Isa|Islay|Jamal|Jayce|Jean|Jody|Jona|Junis|Kaj|Kennedy|Kim|Kourtney|Kyler|Lee|Lindsay|Lou|Luka|Mans|Marley|Meredith|Michele|Mika|Nicola|Nikita|Noah|Oakley|Oyama|Parker|Paulin|Pim|Quincy|Regis|René|Robby|Rory|Rowan|Sasha|Scout|Siro|Shiloh|Taylor|Tiam|Tony|Tyler|Ude|Vada|Valo|Wally|Whitney|Wynne|Zero|Zohar</de>
		</firstname1>
		<firstname2>
			<de>Alexis|Angel|Aria|Ashley|Aydin|Bent|Billy|Blair|Bobby|Camilie|Casey|Chaska|Chris|Dakota|Dany|Deniz|Dylan|Elis|Elia|Enid|Erin|Evin|Feng|Finn|Fiore|Florin|Gates|Gerit|Güven|Hadley|Helge|Hilal|Ilter|Indiana|Ino|Ira|Jade|Jamie|James|Jayden|Jesse|Jorah|June|Kai|Kaya|Kelsey|Kerry|Konni|Kris|Leslie|Loki|Luca|Maria|Marlin|Marlo|Marlon|Maxime|Merle|Milou|Nicki|Nikky|Özbek|Ola|Pahul|Pat|Payden|Pleun|Raleigh|Raven|Reese|Reign|Riley|Robin|Ryan|Sandy|Sean|Shannon|Skyler|Simone|Terry|Tjade|Toni|Umut|Uli|Val|Vanja|Wanja|Yael|Yuki|Zia</de>
		</firstname2>
		<firstname3>
			<de>Amar|Andrea|Ari|Ashton|Bailey|Bene|Bente|Blake|Bo|Bob|Carol|Carter|Charly|Corey|Dani|Darcy|Derya|Dominique|Eike|Elvin|Emerson|Esin|Farina|Fidan|Friedel|Francis|Gin|Gwenaël|Hadar|Haik|Harper|Hemke|Ilya|Indigo|Isa|Jaime|Janne|Jean|Jessi|Jordan|Joyce|Kay|Kendall|Kersten|Kiran|Kristan|Lesley|Lorin|Luan|Malu|Marian|Maxi|Mel|Mian|Nika|Niki|Nino|Nova|Paris|Patrice|Phoenix|Orly|Qian|Quinn|Rayan|Renée|Rey|Robbie|Sam|Sanja|Sascha|Shanti|Sidney|Theodore|Tomke|Tracy|Ulli|Uzay|Valentine|Vesa|Yuri|Wisal|Zisan</de>
		</firstname3>

And I was needing Scandinavian names for my coming update.

		<artistfirstname> <!-- Scandinavian / Nordic name generator doesn't seem to be implemented yet -->
			<de>Lars|Erik|Björn|Henrik|Olaf|Magnus|Anders|Sven|Thor|Gustav|Axel|Finn|Oskar|Kristian|Rune|Nils|Jørgen|Viggo|Arvid|Harald</de>
		</artistfirstname>
		<artistlastname>
			<de>Hansen|Johansson|Andersen|Nielsen|Jørgensen|Svensson|Petersen|Olsen|Larsen|Pedersen|Eriksson|Christiansen|Nilsen|Berg|Magnusson|Kristiansen|Sørensen|Sandberg|Lund|Holm</de>
		</artistlastname>

GWRon commented:
I guess not all languages/countries have "gender neutral" names. Some surely _always_ have suffixes for male/female.

Yet I think an additional table for "Neutral" should/could be added - yet I am not sure how to handle it then for countries without "neutral"-table-content added... randomly choose between male/female?


@ scandinavian
we have Denmark - for now. But Think we could maybe extend the list a bit (depending on what we "need".)

scr0llbaer9000 commented:
> randomly choose between male/female?

that would be yet another parameter then (but I guess redundant because you can always use `|` (OR) in templates.

I meant a parameter that returns a name that would always fit any gender.

GWRon commented:
Yes - but this only works for certain languages/countries. Some countries simply append suffixes to a name to tell you if it is a boy or a girl.

GWRon commented:
https://en.wiktionary.org/wiki/Category:Greek_common-gender_surnames
--- maybe we need similar for other languages.

scr0llbaer9000 commented:
> but this only works for certain languages/countries

ah I mis-/overread that. But I think some of these names I sourced from the internet can be applied for most languages / cultures too as most of them are English or English-sounding or international / universal, and such types of names are somewhat popular in every language / culture, especially in modern times anyway.

nittka commented on source/Dig/base.util.persongenerator.bmx:
```diff
@@ -928,111 +934,252 @@ End Type
 
 '=== GREEK ===
-Type TPersonGeneratorCountry_Greek extends TPersonGeneratorCountry
+rem
```
DIese Änderung in den anderen PR verschieben und hier wirklich nur den polnischen Personengenerator ergänzen?

GWRon replied:
Ich raeume den dann hier nochmal auf bzw mache einen sauberen neuen auf...wenn der griechen-branch uebernommen wurde.

nittka commented on source/Dig/base.util.persongenerator.bmx:
```diff
@@ -928,111 +934,252 @@ End Type
+			"Dawid", "Dariusz", "Damian", "Daniel", "Dominik", ..
+			"Edward", "Emil", ..
+			"Fabian", "Filip", "Franciszek", ..
+			"Grzegorz", ..
```
Here are even more cases, where there is only one name per letter...

nittka reviewed (changes requested):
For my feeling there are too many cases where there is only one first name/last name for a letter.

GWRon commented:
How many family names with X in German do you know?

How many start with Y ? Surely less than with A, E, ...etc.


The names list was based on some wikipedia list and also chatgpt.

Other languages of course might have other less common starting letters for their names...
Adding more rare names there..will lead to more exotic names at the end.

nittka commented:
I am not talking about X and Y but G, H, N, S, T etc. Compared to Greek, there are VERY few first and last names here. There is surely more than one common female first name with H.

GWRon commented:
I extended the Polish-entries according to the forebears.io website so that I now have >100 male, female and familynames (each).

Creating a clean PR for it now.

scr0llbaer9000 commented:
> Other languages of course might have other less common starting letters for their names...

Just as a side note, I recently noticed while adding names that fictional names by novel and screenplay authors are mostly chosen much more "creatively" than real names. They use a lot more quite uncommon names (e.g. first names like Cap, Blondell, Thaddeus), for which there are none or hardly any examples in (my) data of (now) over 5000 real people.

I just wanted to throw that in, but *maybe* there could be a parameter to indicate how common and usual a generated name should be.

GWRon commented:
Within the PR for "Greece" you will find my remarks about exact this issue.
I will create a new issue for this.
